nextcord/application_command.py

```diff
@@ -1265,77 +1305,176 @@ def __init__(
             cmd_arg_given = False
 
         self.name = cmd_arg.name or parameter.name
+        # Use the given name, or default to the parameter name.
+        # typehint_origin = typing.get_origin(parameter.annotation)  # TODO: Once Python 3.10 is standard, use this.
+        typehint_origin = typing_extensions.get_origin(parameter.annotation)
+
+        annotation_type: ApplicationCommandOptionType = MISSING
+        annotation_required = True
+        annotation_choices: List[Union[str, int, float]] = []
+        annotation_channel_types: List[ChannelType] = []
+        annotation_converters: List[OptionConverter] = []
+
+        if typehint_origin is Literal:
+            # If they use the Literal typehint as their base. This currently should only support int, float, str, and
+            #  technically None for setting it to be optional.
+            found_type = MISSING
+            found_choices = []
+            # for lit in typing.get_args(parameter.annotation):  # TODO: Once Python 3.10 is standard, use this.
+            for lit in typing_extensions.get_args(parameter.annotation):
+                lit = unpack_annotated(lit, list(self.option_types.keys()))
+                lit_type = type(lit)
+                if lit is None:
+                    # If None is included, they want it to be optional. But we don't want None added to the choices.
+                    annotation_required = False
+
+                elif lit_type in (int, str, float):
+                    if found_type is MISSING:
+                        # If we haven't set the type of the annotation, set it.
+                        found_type = self.get_type(lit_type)
+                    elif self.get_type(lit_type) is not found_type:
+                        raise ValueError(
+                            f"{self.error_name} | Literal {lit} is incompatible with {found_type}"
+                        )
+                    found_choices.append(lit)
+
+                else:
+                    raise ValueError(
+                        f"{self.error_name} Invalid type for choices: {type(lit)}: {lit}"
+                    )
+
+            if found_type is MISSING:
+                # ??? Why are they using Literal???
```

Comment:
This comment should probably be a little more formal or the code should error out so the user knows not to do this.

Reply:
Alrighty then.